wasm/section.go

```diff
@@ -25,24 +25,82 @@ const (
 	SectionIDData     SectionID = 11
 )
 
+// writeSections appends this module's sections into the buffer in the order required by the specification, custom
+// section first.
+// See https://www.w3.org/TR/wasm-core-1/#modules%E2%91%A0%E2%93%AA
+func (m *Module) encodeSections(buffer []byte) (bytes []byte) {
+	bytes = buffer
+	for name, data := range m.CustomSections {
+		bytes = append(bytes, encodeCustomSection(name, data)...)
+	}
+	if len(m.TypeSection) > 0 {
+		panic("TODO: TypeSection")
```

Comment:
note: I used panic here as we will never need to return an error encoding even if we always do on the decode side